lib/codebreaker.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative 'codebreaker/secret_code'
+require_relative 'codebreaker/validate_service'
+require_relative 'codebreaker/version'
+
+module Codebreaker
+  # Main class for codebreaker game
+  class Main
+    attr_writer :secret_code, :secret_shelter
+    attr_accessor :user_code
+    attr_reader :result
+
+    def initialize
+      @secret_code = []
+      @user_code = []
+      @result = []
+    end
+
+    def play(user_code, secret_code)
```

Comment:
You should store secret code inside class and remove availability to change it outside of class

Reply:
Also u need to generate secret code in initialize method or in separate method